### ISSUE TITLE
unable to compile when using recaptcha plugin

### ISSUE THREAD

**Issue by snimavat** [iamthechad/grails3-recaptcha#17]
Exception occurs when doing grails compile

```
java.lang.RuntimeException: Unable to configure class node for class com.megatome.grails.RecaptchaService due to malformed parameterized types
        at org.codehaus.groovy.vmplugin.v5.Java5.configureClassNode(Java5.java:394)
        at org.codehaus.groovy.ast.ClassNode.lazyClassInit(ClassNode.java:277)
        at org.codehaus.groovy.ast.ClassNode.getUnresolvedSuperClass(ClassNode.java:1006)
        at org.codehaus.groovy.ast.ClassNode.getUnresolvedSuperClass(ClassNode.java:1001)
        at org.codehaus.groovy.ast.ClassNode.getSuperClass(ClassNode.java:995)
        at org.codehaus.groovy.ast.ClassNode.isDerivedFrom(ClassNode.java:938)
```

Grails Version: 3.2.4
Groovy Version: 2.4.7
JVM Version: 1.7.0_79
Gradle 3.0

**Comment by snimavat** [comment]
Same with jdk 8

**Comment by iamthechad** [comment]
I'm having trouble recreating this. I created a new Grails project using Grails 3.2.4 and Gradle 3.0. After adding the `recaptcha` plugin, running `grails compile` works with no errors.

Are there any other errors or warnings? 
Which version of the plugin are you trying to use?

**Comment by snimavat** [comment]
@iamthechad can u merge pr https://github.com/iamthechad/grails3-recaptcha/pull/18 
Even though it complains about coverage, there's nothing changed which would decrease coverage

**Comment by snimavat** [comment]
@iamthechad btw i was trying to use recaptcha 3.0.1 and i am in the process of upgrading recaptcha-spring-security to grails 3 -- https://grails.org/plugin/recaptcha-spring-security

**Comment by iamthechad** [comment]
I see the error in your branch.

I'm working on the Grails version upgrade, but I have deployed a snapshot that works for me when I put it into your code.

Add this to the `repositories`:

    maven { url 'https://oss.jfrog.org/artifactory/libs-snapshot' }

and add the snapshot in dependencies as `compile "org.grails.plugins:recaptcha:3.1.0-SNAPSHOT"`

Hopefully this can get you working temporarily until I can formally deploy a new version.

**Comment by snimavat** [comment]
Great, thanks @iamthechad

**Comment by iamthechad** [comment]
This seems to have been fixed by #19
